streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -1428,13 +1432,23 @@ private long advanceNowAndComputeLatency() {
      */
     public void shutdown() {
         log.info("Informed to shut down");
+
+
+
         final State oldState = setState(State.PENDING_SHUTDOWN);
         if (oldState == State.CREATED) {
             // The thread may not have been started. Take responsibility for shutting down
             completeShutdown(true);
         }
     }
 
+    public void setLeaveGroupOnClose(final boolean leaveGroupOnClose) {
+        if (leaveGroupOnClose) {
+            log.info("Leaving consumer group on close");
+        }
+        ((LegacyKafkaConsumer<?, ?>) mainConsumer).overrideLeaveGroupOnClose();
```

Comment:
If the casting is a acceptable way, we can enhance `LegacyKafkaConsumer#close` to take `CloseOption` right now. `LegacyKafkaConsumer` is not a public class so the change is ok. Also, that would be a compatible way with new feature in the future.